include/settings/settings.h

```diff
@@ -120,6 +121,19 @@ int settings_register(struct settings_handler *cf);
  */
 int settings_load(void);
 
+/**
+ * @brief Load settings that matches selected pattern
+ *
+ * Load all the settings that matches the selected pattern.
+ *
+ * @note The function does not call commit at the end.
```

Comment:
Does it mean that it will be called only once per key with the newest value? (As it should be for every other case...)

Reply:
@pdunaj, No, it will call the handler for every entry it finds that matches the pattern.

Reply:
That means we still need `commit`... application must know what is the most recent data.

Reply:
When the function finishes you are done. I was thinking about this issue but I decided not to call commit after loading as it would complicate the whole thing and maybe you wish to do something else than when system is initialised.

Reply:
I guess it would not be a problem if pattern matches only the entry caller is interested in (preferably in the same module). It may be a different case however if pattern is chosen to trigger reload on a specific set of options coming from various modules. In that case one would have to manually call many commits (I expect commits have internal linkage so this may not be possible). Considering above I think we are doomed to call commit from setting.

Reply:
The commit could be limited to the pattern as well.